libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -799,13 +799,9 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 	}
 	case Type::Category::Function:
 		if (member == "selector")
-		{
-			solUnimplementedAssert(false, "");
-		}
+			define(IRVariable{_memberAccess}, IRVariable(_memberAccess.expression()).part("functionIdentifier"));
```

Comment:
Actually, I think for the ``"selector"`` case, we might have to
```
solUnimplementedAssert(dynamic_cast<FunctionType const&>(*_memberAccess.expression().annotation().type).kind() == FunctionType::Kind::External, "");
```
It also wouldn't hurt to ``solAssert`` the same for the ``"address"`` case.

Otherwise we need to make sure that this works for all kinds of functions, so in particular we need tests for:
```
library L { function f() external {}}
contract C {
 function f() returns(bytes4) { return L.f.selector; }
}
```
and
```
contract A {
  function f() external {}
}
contract C {
 function f() returns(bytes4) { return A.f.selector; }
}
```
and
```
contract A {
  function f() public {}
}
contract C is A {
 function f() returns(bytes4) { return A.f.selector; }
}
```

Reply:
library case fails with:
`solUnimplementedAssert(!_contract.isLibrary(), "Libraries not yet implemented.");`
I guess libraries are not implemented in yul2sol.

Unluckily, both other cases are not running as f is unknown member of A, in both cases.
If this is a bug, I can take a look and try fixing, and if it is not implemented I can't get these tests running. Error code is:
`solAssert(false, "Member access to unknown type.");`

@ekpyron

Reply:
Yeah, apparently we don't have the "member access to type type" in ``A.f`` implemented yet, then - no wonder, since the main reason to do so would be to support their ``A.f.selector`` :-). But as long as you put the assertion that the function call kind is external in here, it's good enough for now and we can deal with the other cases later separately.

Reply:
I added assert at least